### DISCUSSION TITLE
Waterway Relations

### DISCUSSION THREAD

**Discussion by georgbachmann** [osm2pgsql-dev/osm2pgsql#1694] [General]
I though I have the same problem before reading [here](https://github.com/openstreetmap/osm2pgsql/issues/1545) :) But my question is the following:
I use osm2pqsql to import OSM data to later on export it to display a map. I want to use information like wikipedia and wikidata and in the case of e.g. rivers (e.g. https://www.openstreetmap.org/relation/89652) this information is on the relation, but not on the individual LineStrings that make out the whole relation.

So my question is... how would I address that? How do I get the relation inside my database?

**Comment by joto** [comment]
Read the [chapter about 2 stage processing](https://osm2pgsql.org/doc/manual.html#stages) in the documentation. This can be a bit tricky to get right though. The other option would be to import the relations into an extra table without geometry, just to store the member ids and the wikipedia/wikidata tags you need and then join the data in the database with the member nodes and ways.

**Reply by georgbachmann** [reply]
Thanks @joto for your reply and your help!
So I think there should be a `planet_osm_rels` table, right? For me it did not create that table?!? Not sure why... Our system should have plenty (about 250GB) of RAM... 
I could imagine that this table could solve my problem like you described. Do you have any idea what I have to do different to get that table? I just saw that I already asked about that table [here](https://gis.stackexchange.com/questions/288784/how-to-import-osm-relations-to-postgresql-databases) but never got a response and then forgot about it it seems ;) 
Would be great if you have an idea...

**Reply by joto** [reply]
No. The `planet_osm_rels` table is for updating when `--slim` mode is used, basically for internal use of osm2pgsql. Don't use that, because its format might change. Use the flex output and create your own table with just the information you need.

**Reply by pnorman** [reply]
You can see something similar for the route table in https://github.com/gravitystorm/openstreetmap-carto/pull/4431

It imports routes, not waterway relations, but the idea is similar.
